packages/react/src/components/DetailsList/DetailsColumn.base.tsx

```diff
@@ -129,11 +129,14 @@ export class DetailsColumnBase extends React.Component<IDetailsColumnProps> {
                       <IconComponent className={classNames.iconClassName} iconName={column.iconName} />
                     )}
 
-                    {column.isIconOnly ? (
-                      <span className={classNames.accessibleLabel}>{column.name}</span>
-                    ) : (
-                      column.name
-                    )}
+                    {column.onRenderHeader && column.onRenderHeader()}
+
+                    {!column.onRenderHeader &&
```

Comment:
This should be the `defaultRender` behavior in `onRenderHeader`.

Reply:
This should be fixed now.